GVFS/GVFS/CommandLine/UpgradeVerb.cs

```diff
@@ -0,0 +1,309 @@
+                    {
+                        upgradeInstallable = true;
+
+                        if (!this.TryCheckUpgradeInstallable(out errorMessage))
+                        {
+                            upgradeInstallable = false;
+                        }
+                    }
+
+                    return upgradeCheckSuccess;
+                },
+                 "Checking for GVFS upgrades",
+                suppressGvfsLogMessage: true);
+
+            latestVersion = version;
+            installable = upgradeInstallable;
+            error = errorMessage;
+
+            return upgradeCheckSuccess;
+        }
+
+        private bool TryRunInstaller(out string error)
+        {
+            string upgraderPath = null;
+            string errorMessage = null;
+
+            bool preUpgradeSuccess = this.ShowStatusWhileRunning(
+                () =>
+                {
+                    if (this.TryCopyUpgradeTool(out upgraderPath, out errorMessage) &&
+                        this.TryLaunchUpgradeTool(upgraderPath, out errorMessage))
+                    {
+                        return true;
+                    }
+
+                    return false;
+                },
+                "Launching upgrade tool",
+                suppressGvfsLogMessage: true);
+            
+            if (!preUpgradeSuccess)
+            {
+                error = errorMessage;
+                return false;
+            }
+
+            error = null;
+            return true;
+        }
+
+        private bool TryCopyUpgradeTool(out string upgraderExePath, out string error)
+        {
+            upgraderExePath = null;
+
+            this.tracer.RelatedInfo("Copying upgrade tool");
+
+            if (!this.upgrader.TryCreateToolsDirectory(out upgraderExePath, out error))
+            {
+                return false;
+            }
+
+            this.tracer.RelatedInfo("Successfully Copied upgrade tool.");
+
+            return true;
+        }
+
+        private bool TryLaunchUpgradeTool(string path, out string error)
+        {
+            this.tracer.RelatedInfo("Launching upgrade tool");
+
+            if (!this.processWrapper.Start(path))
+            {
+                error = "Error launching upgrade tool";
+                return false;
+            }
+            
+            this.tracer.RelatedInfo("Successfully launched upgrade tool.");
+
+            error = null;
+            return true;
+        }
+
+        private bool TryCheckUpgradeAvailable(
+            out Version latestVersion, 
+            out string error)
+        {
+            latestVersion = null;
+            error = null;
+
+            this.tracer.RelatedInfo("Checking server for available upgrades.");
+
+            bool checkSucceeded = false;
+            Version version = null;
+
+            checkSucceeded = this.upgrader.TryGetNewerVersion(out version, out error);
+            if (!checkSucceeded)
+            {
+                return false;

```

Comment:
Log warning\error here?

(This comment applies in all of the `TryXXX` methods that log when they succeed but not when they fail)

Reply:
The failures are getting logged by the caller method - ```TryRunProductUpgrade```.